tests/unit/test_main.py

```diff
@@ -1178,39 +1184,40 @@ def main_check(args):
         "/git_project1/nested_dir/has_imports.py",
     ]
 
-    out, error = main_check([str(tmpdir), "--skip-gitignore", "--filter-files", "--check"])
+    out, error = main_check([str(tmp_path), "--skip-gitignore", "--filter-files", "--check"])
 
     if os.name == "nt":
         should_check = [sc.replace("/", "\\") for sc in should_check]
 
-    assert all(f"{str(tmpdir)}{file}" in error for file in should_check)
+    assert all(f"{str(tmp_path)}{file}" in error for file in should_check)
 
-    out, error = main_check([str(tmpdir), "--skip-gitignore", "--filter-files"])
+    out, error = main_check([str(tmp_path), "--skip-gitignore", "--filter-files"])
 
-    assert all(f"{str(tmpdir)}{file}" in out for file in should_check)
+    assert all(f"{str(tmp_path)}{file}" in out for file in should_check)
 
     # Should work when git project contains symlinks
 
     if os.name != "nt":
-        git_project0.join("has_imports_ignored.py").write(import_content)
-        git_project0.join("has_imports.py").write(import_content)
-        tmpdir.join("has_imports.py").write(import_content)
-        tmpdir.join("nested_dir").join("has_imports.py").write(import_content)
-        git_project0.join("ignore_link.py").mksymlinkto(tmpdir.join("has_imports.py"))
-        git_project0.join("ignore_link").mksymlinkto(tmpdir.join("nested_dir"))
-        git_project0.join(".gitignore").write("ignore_link.py\nignore_link", mode="a")
+        git_project0.joinpath("has_imports_ignored.py").write_text(import_content)
+        git_project0.joinpath("has_imports.py").write_text(import_content)
+        tmp_path.joinpath("has_imports.py").write_text(import_content)
+        tmp_path.joinpath("nested_dir").joinpath("has_imports.py").write_text(import_content)
+        git_project0.joinpath("ignore_link.py").symlink_to(tmp_path.joinpath("has_imports.py"))
+        git_project0.joinpath("ignore_link").symlink_to(tmp_path.joinpath("nested_dir"))
+        gitignore = git_project0.joinpath(".gitignore")
+        gitignore.write_text(gitignore.read_text() + "ignore_link.py\nignore_link")
 
         out, error = main_check(
             [str(git_project0), "--skip-gitignore", "--filter-files", "--check"]
         )
 
         should_check = ["/git_project0/has_imports.py"]
 
-        assert all(f"{str(tmpdir)}{file}" in error for file in should_check)
+        assert all(f"{str(tmp_path)}{file}" in error for file in should_check)
```

Comment:
Using both `str` and `f"` shouldn't be necessary for `Path`

Reply:
> Using both `str` and `f"` shouldn't be necessary for `Path`

what do you think about making this a separate change: I see there are several points in this file (including parts not touched by this change) that could do with the same update? I'm happy to include it in this patch, but my main goal was to just get things working, without changing things too much

Reply:
Is it possible to exclude the `py` change and do that together with the changes in a separate PR?

If not, let's at least make sure the code we're touching now in this diff is correct and worry about the rest later.

Reply:
> If not, let's at least make sure the code we're touching now in this diff is correct and worry about the rest later.

:+1: [830eca7](https://github.com/PyCQA/isort/pull/2329/commits/830eca7567e136d146e5a1e8dd33d81f1459628e)